examples/simple-react-app/src/App.jsx

```diff
@@ -0,0 +1,68 @@
+import { useState } from 'react';
+import useDCPWorker from 'use-dcp-worker';
+
+const workerPaymentAddress = import.meta.env.VITE_WORKER_PAYMENT_ACCOUNT;
+
+function App() {
+  const [paymentAddress, setPaymentAddress] = useState(workerPaymentAddress);
+  // window.dcp is populated from dcp-client script in public/index.html 
+  const { wallet } = window.dcp;
+
+  // triggers client modal and sets workerPaymentAddress
+  async function getPaymentAddress()
+  {
+    console.warn('Did not specifiy BANK_ACCOUNT_ADDRESS env var. Calling wallet.get.');
+    try
+    {
+      let ks = await wallet.get(); // Wallet API
+      setPaymentAddress(ks.address);
+    }
+    catch (e)
+    {
+      console.error(e);
+      alert(e);
+    }
+  }
+
+  // resolve payment address for worker options
+  // if .env var is not set, trigger client modal
+  if (!workerPaymentAddress)
+    getPaymentAddress();
```

Comment:
**suggestion:** Use the state variable rather than the initial value, otherwise `getPaymentAddress()` will be called on every render if the environment variable isn't set.

```suggestion
  if (!paymentAddress)
    getPaymentAddress();
```